### DISCUSSION TITLE
Using pocketbase client in multiple threads

### DISCUSSION THREAD

**Discussion by TomFreudenberg** [pocketbase/pocketbase#4185] [Q&A] [answered]
When starting a number of threads in an app, is it safe to instantiate only one pbclient connection and share that to the threads or must every thread gets its own pbclient connection? Requests in threads should be run in parallel.

Is the client in general thread safe?

Thanks for a short help!
Tom

**Comment by ganigeorgiev** [comment]
What do you mean by "pbclient" (_which client?_) and "starting a number of threads in an app" (_what app?_)?

Could you elaborate a little more on your use case and in what environment your app is running?

**Reply by ganigeorgiev** [reply, marked as answer]
Note that PocketBase web APIs are fully stateless and there are no sessions in the traditional sense.
A single request is considered authenticated if it has a valid `Authorization:TOKEN` header with it.
This header is added automatically by the SDKs (JS and Dart) if the respective `pb.authStore.token` is nonempty.

If by "pbclient" you mean the JS or Dart SDK client instance, then whether it would be safe to have a single shared instance (_if that is what you are asking_) depends whether you are processing requests from different users because the `pb.authStore` will be shared and requests from different users will write over the same object. This is why for JS in SSR context is recommended for the regular user auth to use one client per request (see more [JS SDK SSR integration](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#ssr-integration)).

**Reply by TomFreudenberg** [reply]
Hey, pretty nice your answer


I am looking for backend Jobs. I have built an App based on Event Driven Architecture and therefor are tasks on message queue. The worker engine is multiprocessing and multi threading enabled. So a queue can be driven by multiple executers.

Therefor it is less resource consuming to use a shared web api object.

I use the pbclient library for Python.
